internal/oracle/contracts.go

```diff
@@ -0,0 +1,111 @@
+package oracle
+
+import (
+	"context"
+	"encoding/hex"
+	"strings"
+
+	"github.com/tokenized/identity-oracle/internal/platform/db"
+	"github.com/tokenized/pkg/bitcoin"
+	"github.com/tokenized/pkg/storage"
+	"github.com/tokenized/specification/dist/golang/actions"
+	"github.com/tokenized/specification/dist/golang/protocol"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// contractsStorageKey is the path to the contract formations.
+	contractsStorageKey = "contract_formations"
+)
+
+// ContractsManager implements the ContractProcessor interface for the contracts package. It saves
+// contract formation actions to storage.
+type ContractsManager struct {
+	st     storage.Storage
+	isTest bool
+}
+
+// NewContractsManager creates a ContractsManager.
+func NewContractsManager(st storage.Storage, isTest bool) *ContractsManager {
+	return &ContractsManager{st, isTest}
+}
+
+// SaveContractFormation saves a contract formation to storage.
+func (cm *ContractsManager) SaveContractFormation(ctx context.Context, ra bitcoin.RawAddress,
+	script []byte) error {
+
+	key := strings.Join([]string{contractsStorageKey, hex.EncodeToString(ra.Bytes())}, "/")
+
+	// Check for pre-existing
+	b, err := cm.st.Read(ctx, key)
+	if err == nil {
+		// Check timestamp vs current version to ensure we keep the latest.
```

Comment:
De-nest this main block.

Reply:
I think we just need to be careful to not prioritize statement depth of logic complexity. It is difficult for me because I have always thought that positive if statements are simpler, but golang seems to want a lot more negative if statements (if not error). I subconsciously default to the if "condition I want" statement, which I think has merit. I have changed this code though because it is a 3 way if statement that is difficult to read in any form. Just something to think about.